File: android/app/src/main/java/com/aitutor/app/domain/model/Schedule.kt
package com.aitutor.app.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class StudySession(
    val id: String,
    val title: String,
    val notes: String? = null,
    @SerialName("scheduled_for") val scheduledFor: String, // ISO-8601 timestamp
    @SerialName("user_id") val userId: String,
    @SerialName("created_at") val createdAt: String? = null
)

File: ios/AITutor/Domain/Model/Schedule.swift
import Foundation

struct StudySession: Codable, Equatable, Identifiable {
    let id: String
    let title: String
    let notes: String?
    /// ISO-8601 timestamp.
    let scheduledFor: String
    let userId: String
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, title, notes
        case scheduledFor = "scheduled_for"
        case userId = "user_id"
        case createdAt = "created_at"
    }
}
